src/expressions/method-call-expr.md

```diff
@@ -23,7 +23,18 @@ Then, for each candidate `T`, add `&T` and `&mut T` to the list immediately afte
 
 For instance, if the receiver has type `Box<[i32;2]>`, then the candidate types will be `Box<[i32;2]>`, `&Box<[i32;2]>`, `&mut Box<[i32;2]>`, `[i32; 2]` (by dereferencing), `&[i32; 2]`, `&mut [i32; 2]`, `[i32]` (by unsized coercion), `&[i32]`, and finally `&mut [i32]`.
 
-Then, for each candidate type `T`, search for a [visible] method with a receiver of that type in the following places:
+A second - possible more expansive - list is then made of types where we might
+find [visible] methods with a receiver of that type, called the list of
+contributing types. To make this list, follow the exact same process,
+but this time use the `Receiver` trait
+instead of the `Deref` trait for any non-built-in derefences.
+
+There is a blanket implementation of `Receiver` for all `T: Deref`, so in many
+cases the lists are identical, but there are rare smart pointers which
+may implement `Receiver` without implementing `Deref`, so this second list of
+types may be longer than the list of candidate types.
```

Comment:
Is there any reason that we shouldn't just merge this into the description further above and describe only a single list, or is there some semantic importance to describing this as one list being created, and then another possibly more expansive one being created?

Reply:
I've had a crack at this in c860df3422ddf2c3430ed95e7b6e25e889186bf0. See what you think.

Reply:
...continued in https://github.com/rust-lang/reference/pull/1699#discussion_r1932767574.